presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -150,8 +170,100 @@ public void testColumnsCached()
+        assertThat(cachingJdbcClient.getColumns(secondSession, table)).doesNotContain(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 2, 4);
+    }
+
+    @Test
+    public void testColumnsCacheInvalidationOnTableDrop()
+    {
+        ConnectorSession firstSession = createSession("first");
+        ConnectorSession secondSession = createSession("second");
+        JdbcTableHandle firstTable = createTable(new SchemaTableName(schema, "first_table"));
+        JdbcTableHandle secondTable = createTable(new SchemaTableName(schema, "second_table"));
+
+        JdbcColumnHandle firstColumn = addColumn(firstTable, "first_column");
+        JdbcColumnHandle secondColumn = addColumn(secondTable, "second_column");
+
+        // Load columns for both tables into cache and assert cache loads (sessions x tables)
+        assertThat(cachingJdbcClient.getColumns(firstSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(firstSession, secondTable)).contains(secondColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 0, 4);
+
+        // Load columns from cache
+        assertThat(cachingJdbcClient.getColumns(firstSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 2, 4);
+
+        // Rename column
+        cachingJdbcClient.renameColumn(firstSession, firstTable, firstColumn, "another_column");
+        assertThat(cachingJdbcClient.getColumns(secondSession, firstTable))
+                .doesNotContain(firstColumn)
+                .containsAll(jdbcClient.getColumns(SESSION, firstTable));
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 5, 2, 5);
+
+        // Drop columns and caches for first table
+        cachingJdbcClient.dropTable(secondSession, firstTable);
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(firstSession, firstTable)).isInstanceOf(TableNotFoundException.class);
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(secondSession, firstTable)).isInstanceOf(TableNotFoundException.class);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 7, 2, 7);
+
+        // Check if second table is still cached
+        assertThat(cachingJdbcClient.getColumns(firstSession, secondTable)).contains(secondColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 7, 4, 7);
```

Comment:
please consider extracting variables `Loads`, `Hits`, `Misses` and only incrementing here
`assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), loads, hits += 2, misses);`
so no need to search for previous state and one can concentrate only on state change.

Reply:
These values are so small that I don't see values in using variables here

Reply:
After a short consideration, I gave it a shot and it looks better. Thanks!